packages/helix-shared-config/src/ValidationError.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-disable max-classes-per-file */
+
+class ValidationError extends Error {
+  constructor(msg, errors = [], mapError = ValidationError.mapError) {
+    const detail = errors.map((e) => mapError(e)).join('\n');
+    super(`Invalid configuration:
```

Comment:
I saw no other way to add a class-specific custom error mapping, than passing it as argument; I can't add a instance method of that name, as it can not be invoked in the constructor itself with `this`.